appveyor.yml

```diff
@@ -38,14 +43,10 @@ build_script:
 - call C:\PX4\toolchain\scripts\setup-environment.bat x
 # safe the repopath for switching to it in cygwin bash
 - for /f %%i in ('cygpath -u %%CD%%') do set repopath=%%i
-# fetch all submodules in parallel
-- call bash --login -c "cd $repopath && git submodule -q update --init --recursive --jobs=10"
-# make SITL
-- call bash --login -c "cd $repopath && make px4_sitl_test"
-# make pixracer to check NuttX build
-- call bash --login -c "cd $repopath && make px4_fmu-v4_default"
-# run tests
-- call bash --login -c "cd $repopath && make tests"
+# fetch all submodules in parallel with limited depth
+- call bash --login -c "cd $repopath && git submodule -q update --init --recursive --jobs=10 --depth 200"
```

Comment:
I'm not sure if limiting to a depth of 200 will always work.

Reply:
Have you tried skipping the submodule update entirely? Letting the build system handle it will mean the sitl appveyor build doesn't even bother cloning nuttx.

Reply:
Would make a lot of sense, I didn't know that's supported.

Reply:
I took the 200 depth from your pr: https://github.com/PX4/Firmware/pull/10865/files#diff-180360612c6b8c4ed830919bbb4dd459R11 I thought that's what you wanted to achieve and I didn't have any submodule depth before to make it not accidentally fail. Let's make the build system handle it. I tried, it works but there's no additional time-saving.

Reply:
That's the depth of the main repo and it just needs to go back far enough to find a tag (probably should be deeper actually).

Submodules are a little weirder because they're cloned first, then switched to the specific commit. It's possible (although apparently fine in current master) that we'll have our submodules on commits that are further than 200 commits back.

Reply:
> Let's make the build system handle it. I tried, it works but there's no additional time-saving.

There's value in doing nothing extra to make sure the automatic submodule initialization handled by cmake works on windows/cygwin.

Reply:
> That's the depth of the main repo and it just needs to go back far enough to find a tag (probably should be deeper actually).

The main repo is already cloned with depth one for the necessary branch or tag: https://github.com/PX4/Firmware/blob/43acfc8aa069658f9d18735d67a7da3946e142de/appveyor.yml#L4

> Submodules are a little weirder because they're cloned first, then switched to the specific commit.

GitHub just doesn't support fetching arbitrary commit hashes which are untagged because there's a weakness in the protocol and it would need to advertise every available commit to the fetching client which doesn't scale. So for the submodules shallow clone with depth one from within git doesn't work but downloading the tree at an arbitrary commit would actually be supported e.g. https://github.com/PX4/Firmware/archive/88f2a6854c6e46a235c3c5959fc8f6f56408d932.zip